src/ast/effects.h

```diff
@@ -67,6 +69,9 @@ struct EffectAnalyzer : public PostWalker<EffectAnalyzer> {
                  || (accessesMemory() && (other.writesMemory || other.calls))) {
       return true;
     }
+    // All atomics are sequentially consistent for now, but have no ordering
+    // constraints wrt non-atomics.
+    if (isAtomic && other.isAtomic) return true;
```

Comment:
based on the discussion, it seems like this should be: if one of the two is atomic, and the other accesses memory (atomically or not, but not sure if both loads and stores?), then return true.